File: app/src/main/java/com/doubean/ford/ui/common/LoadMoreState.java
package com.doubean.ford.ui.common;

public class LoadMoreState {
    private final boolean running;
    private final String errorMessage;
    private boolean handledError = false;

    public LoadMoreState(boolean running, String errorMessage) {
        this.running = running;
        this.errorMessage = errorMessage;
    }

    public boolean isRunning() {
        return running;
    }

    String getErrorMessage() {
        return errorMessage;
    }

    public String getErrorMessageIfNotHandled() {
        if (handledError) {
            return null;
        }
        handledError = true;
        return errorMessage;
    }
}

File: app/src/main/java/com/doubean/ford/ui/common/LoadMoreState.kt
package com.doubean.ford.ui.common

class LoadMoreState(val isRunning: Boolean, val errorMessage: String?) {
    private var handledError = false
    val errorMessageIfNotHandled: String?
        get() {
            if (handledError) {
                return null
            }
            handledError = true
            return errorMessage
        }
}